raven/src/main/java/org/opencadc/raven/ServiceAvailability.java

```diff
@@ -0,0 +1,186 @@
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+ */
+
+package org.opencadc.raven;
+
+import ca.nrc.cadc.auth.AuthMethod;
+import ca.nrc.cadc.db.DBUtil;
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.reg.client.LocalAuthority;
+import ca.nrc.cadc.reg.client.RegistryClient;
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.vosi.AvailabilityPlugin;
+import ca.nrc.cadc.vosi.AvailabilityStatus;
+import ca.nrc.cadc.vosi.avail.CheckException;
+import ca.nrc.cadc.vosi.avail.CheckResource;
+import ca.nrc.cadc.vosi.avail.CheckWebService;
+
+import java.net.URI;
+import java.util.Map;
+
+import javax.sql.DataSource;
+
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.version.InitDatabase;
+
+/**
+ * This class performs the work of determining if the executing locate
+ * service is operating as expected.
+ * 
+ * @author majorb
+ */
+public class ServiceAvailability implements AvailabilityPlugin {
+
+    private static final Logger log = Logger.getLogger(ServiceAvailability.class);
+
+    /**
+     * Default, no-arg constructor.
+     */
+    public ServiceAvailability() {
+    }
+
+    /**
+     * Sets the name of the application.
+     */
+    @Override
+    public void setAppName(String string) {
+        //no-op
+    }
+
+    /**
+     * Performs a simple check for the availability of the object.
+     * @return true always
+     */
+    @Override
+    public boolean heartbeat() {
+        return true;
+    }
+
+    /**
+     * Do a comprehensive check of the service and it's dependencies.
+     * @return Information of the availability check.
+     */
+    @Override
+    public AvailabilityStatus getStatus() {
+        boolean isGood = true;
+        String note = "service is accepting requests";
+        
+        try {
+            
+            log.info("init database...");
+            DataSource ds = DBUtil.findJNDIDataSource(PostAction.JNDI_DATASOURCE);
+            MultiValuedProperties props = PostAction.readConfig();
+            Map<String, Object> config = PostAction.getDaoConfig(props);
+            InitDatabase init = new InitDatabase(ds, (String) config.get("database"), (String) config.get("schema"));
```

Comment:
raven should not init the database - it only reads from it; the global database will be created/maintained by metadata-sync